codec/src/test/java/io/netty/handler/codec/compression/JZlibTest.java

```diff
@@ -26,4 +28,10 @@ protected ZlibEncoder createEncoder(ZlibWrapper wrapper) {
     protected ZlibDecoder createDecoder(ZlibWrapper wrapper) {
         return new JZlibDecoder(wrapper);
     }
+
+    @Test(expected = DecompressionException.class)
+    @Override
+    public void testZLIB_OR_NONE3() throws Exception {
```

Comment:
Both work the same way with the current changes (before this they didn't!). Both do not handle gzip compression when the JZlibdecoder and JdkZlibDecoders are used with ZLIB_OR_NONE wrapper type. Is this behavior okay?

The other way is to make both these decoders handle gzip compression when the wrapper type is ZLIB_OR_NONE. 

Please let me know which way to choose.